src/libhirte/common/cfg.c

```diff
@@ -112,21 +112,24 @@ int cfg_load_complete_configuration(
         if (custom_config_file != NULL) {
                 result = cfg_load_from_file(config, custom_config_file);
                 if (result < 0) {
-                        fprintf(stderr,
-                                "Error loading custom configuration file '%s': '%s'.\n",
-                                custom_config_file,
-                                strerror(-result));
                         if (result != -ENOENT) {
+                                fprintf(stderr,
+                                        "Error: loading custom configuration file '%s' failed: '%s'.\n",
+                                        custom_config_file,
+                                        strerror(-result));
                                 return result;
                         }
+                        fprintf(stderr,
+                                "Did not find a custom configuration file '%s'. Skipping.\n",
+                                custom_config_file);
                 }
         }
 
         if (custom_config_directory != NULL) {
                 result = cfg_load_from_dir(config, custom_config_directory);
                 if (result < 0) {
                         fprintf(stderr,
-                                "Error loading configuration from conf.d dir '%s': '%s'.\n",
+                                "Error: loading configuration from conf.d dir '%s' failed: '%s'.\n",
```

Comment:
I think this needs an ENOENT check and report nothing as well.

If I have a /etc/hirte/hirte.conf.d directory with nothing in it, hirte-agent should nor write to the log files.

Reply:
What happens if the directory does exist but is empty.

Reply:
In that case, `scandir` should returns 0 items so that the following for-loop isn't executed and we simply exit the function.

Reply:
Hi @engelmi thanks for catching this one so fast.
I would do different approach, pardon my quick C-foo language below. I would need few days to submit a patch for this one. I have some others tasks around.

@engelmi @rhatdan WDYT:

**cfg.h**
// Please note the config files listed are the order of check/priority.
`#define HIRTE_CONFIG ("/etc/hirte/hirte.conf" "/etc/hirte/hirte.conf.d/hirte.conf" "/usr/share/hirte/config/hirte.conf")`

**cfg.c**

1. remove a kind of multiple functions doing the same or similar:
```
int cfg_load_complete_configuration(
  <snip>
  if (default_config_file != NULL) {
                result = cfg_load_from_file(config, default_config_file);
                if (result < 0) {
                        fprintf(stderr,
                                "Error loading default configuration file '%s': '%s'.\n",
                                default_config_file,
                                strerror(-result));
                        return result;
                }
        }

        if (custom_config_file != NULL) {
                result = cfg_load_from_file(config, custom_config_file);
                if (result < 0) {
                        fprintf(stderr,
                                "Error loading custom configuration file '%s': '%s'.\n",
                                custom_config_file,
                                strerror(-result));
                        if (result != -ENOENT) {
                                return result;
                        }
                }
        }

        if (custom_config_directory != NULL) {
                result = cfg_load_from_dir(config, custom_config_directory);
                if (result < 0) {
                        fprintf(stderr,
                                "Error loading configuration from conf.d dir '%s': '%s'.\n",
                                custom_config_directory,
                                strerror(-result));
                        return result;
                }
        }
```

4. Do a for loop into `HIRTE_CONFIG` define trying to read the files listed there. Also keep a temporary var which will be a list with the config files already tried.

Example:
```
int cfg_load_complete_configuration(
    configfiles_notfound = ARRAY
    configfile_loaded = ""
    for loop into HIRTE_CONFIG:
                result = cfg_load_from_file(config, loop[i]);
                if (result > 0) {
                      return result
                }
                configfile_notfound.append(loop)
   }
   fprintf("Unable to load configuration file, tried the following: %s. Exiting...", configfile_notfound)
   return configfile_notfound

```

Reply:
The same apply to agent config.

Reply:
Hi @dougsland, 
in each case, the return value when loading the configuration file is handled a bit differently. For example, an error when loading the default config will always result in exiting the program while a "Not Found" error when loading the custom config will be "ignored". Also, conf.d might contain multiple files. So unifying this flow most likely results in a code that is more complex and harder to read.